learn2track/models/layers.py

```diff
@@ -58,8 +61,11 @@ def initialize(self, weights_initializer=initer.UniformInitializer(1234)):
     def parameters(self):
         return [self.W, self.b, self.g]
 
-    def fprop(self, X):
-        units_inputs = T.dot(X, self.W)
+    def fprop(self, X, dropout_W=None):
+        W = self.W
+        if dropout_W:
+            W *= dropout_W
```

Comment:
Isn't it `X` that is supposed to the dropped in Dropout? Right now, you are independently dropping the weights of `W`. I think this will cause the hidden units within a layer to not see the information coming from the previous layer.